plugins/copyonwrite/copyonwrite_darwin copy.go

```diff
@@ -0,0 +1,9 @@
+//go:build darwin
+
+package copyonwrite
+
+import "golang.org/x/sys/unix"
+
+func CopyOnWrite(src, dst string) error {
+	return unix.Clonefile(src, dst, 0)
```

Comment:
This will also fail if:

- The filesystem doesn't support copy-on-write.
- The source file is on a different filesystem than the destination.

You can check for both of these ahead of time for every file, but it's more platform-specific logic and overhead. It wouldn't be possible to only do a single check on the original source and destination given to `Copy` either, since a directory can contain mount points for other filesystems.

I think falling back to read and write copy on error is going to be the most compatible approach.